http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/WebSocketSourceHandler.java

```diff
@@ -0,0 +1,183 @@
+        sendOnOpenMessage(ctx, isSecured, uri);
+    }
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws UnknownWebSocketFrameTypeException {
+        cMsg = null;
+        if (!(msg instanceof WebSocketFrame)) {
+            LOGGER.error("Expecting WebSocketFrame. Unknown type.");
+            throw new UnknownWebSocketFrameTypeException("Expecting WebSocketFrame. Unknown type.");
+        }
+        if (msg instanceof TextWebSocketFrame) {
+            TextWebSocketFrame textWebSocketFrame = (TextWebSocketFrame) msg;
+            String text = textWebSocketFrame.text();
+            cMsg = new TextCarbonMessage(text);
+
+        } else if (msg instanceof BinaryWebSocketFrame) {
+            BinaryWebSocketFrame binaryWebSocketFrame = (BinaryWebSocketFrame) msg;
+            boolean finalFragment = binaryWebSocketFrame.isFinalFragment();
+            ByteBuf byteBuf = binaryWebSocketFrame.content();
+            ByteBuffer byteBuffer = byteBuf.nioBuffer();
+            cMsg = new BinaryCarbonMessage(byteBuffer, finalFragment);
+
+        } else if (msg instanceof CloseWebSocketFrame) {
+            CloseWebSocketFrame closeWebSocketFrame = (CloseWebSocketFrame) msg;
+            String reasonText = closeWebSocketFrame.reasonText();
+            int statusCode = closeWebSocketFrame.statusCode();
+            ctx.channel().close();
+            cMsg = new StatusCarbonMessage(org.wso2.carbon.messaging.Constants.STATUS_CLOSE, statusCode, reasonText);
+
+        } else if (msg instanceof PongWebSocketFrame) {
+            //Control message for WebSocket is Pong Message
+            PongWebSocketFrame pongWebSocketFrame = (PongWebSocketFrame) msg;
+            boolean finalFragment = pongWebSocketFrame.isFinalFragment();
+            ByteBuf byteBuf = pongWebSocketFrame.content();
+            ByteBuffer byteBuffer = byteBuf.nioBuffer();
+            cMsg = new ControlCarbonMessage(byteBuffer, finalFragment);
+            byteBuf.release();
+
+        }
+
+        setupCarbonMessage(ctx);
+        publishToMessageProcessor(cMsg);
+    }
+
+
+    /*
+    Carbon Message is published to registered message processor
+    Message Processor should return transport thread immediately
+     */
+    @Override
+    protected void publishToMessageProcessor(CarbonMessage cMsg) {
+        if (HTTPTransportContextHolder.getInstance().getHandlerExecutor() != null) {
+            HTTPTransportContextHolder.getInstance().getHandlerExecutor().executeAtSourceRequestReceiving(cMsg);
+        }
+
+        CarbonMessageProcessor carbonMessageProcessor = HTTPTransportContextHolder.getInstance()
+                .getMessageProcessor();
+        if (carbonMessageProcessor != null) {
+            try {
+                carbonMessageProcessor.receive(cMsg, new ResponseCallback(this.ctx));
+            } catch (Exception e) {
+                LOGGER.error("Error while submitting CarbonMessage to CarbonMessageProcessor.", e);
```

Comment:
I'm also confused with them. Since there are various kinds of ways that people have done it. But since it is a **private static final Logger** isn't it a kind of constants and shouldn't we use the common methodology?